base/primes.jl

```diff
@@ -83,7 +83,7 @@ function primes(lo::Int, hi::Int)
     lo ≤ 3 ≤ hi && push!(list, 3)
     lo ≤ 5 ≤ hi && push!(list, 5)
     hi < 7 && return list
-    sizehint!(list, floor(Int, hi / log(hi)))
+    sizehint!(list, floor(Int, max(hi,0) / log(max(hi,2)) - max(lo,0) / log(max(lo,2))))
```

Comment:
I guess we can clean this up a bit by writing `if lo < 2; lo = 2; end` after the `lo ≤ hi` check. And then just write `hi / log(hi) - lo / log(lo)` for the sizehint